Templates and single binary

calj opened the pull request:
- Template the project with project name, organization and git repository domain
- Create a single binary with server and cli commands

calj committed:
Template the project with project name, organization and git repository domain

calj committed:
Create a single binary with server and cli commands

mod commented on generators/app/index.js:
```diff
@@ -28,17 +42,37 @@ module.exports = class extends Generator {
+      cwd: this.destinationPath()
+    });
+
+    this.spawnCommandSync("git", ["add", "."], {
```
I suggest make this a function and we make git optional
also better we make sure it builds before commiting ?

calj replied:
Added --nogit argument
Then I think we can keep it simple, if the result is not perfect the operator will fix and eventually PR to the generator.

mod commented on generators/app/templates/cmd/sonic/sonic.go:
```diff
@@ -4,7 +4,10 @@ import (
+  "github.com/foolin/goview/supports/ginview"
+	"github.com/gin-gonic/gin"
+	"<%= git_domain %>/<%= organization %>/<%= project %>/config"
+	"<%= git_domain %>/<%= organization %>/<%= project %>/routes"
```
I suggest making a function or simply a prop : <%= package_name %>
project name being the last element of the url

calj replied:
done

mod reviewed (changes requested):
very interesting start, we can merge but I feel it's good to comment more here and collaborate

calj committed:
Add optional arguments to the generator to give the ability to use it in scripts
Add gopkg variable to templates to simplify the code in template
Add ability to disable git with --nogit argument
